### PR TITLE
JBPM-8682 Process Instance List basic filter checkbox are not working…

### PR DESCRIPTION
**Pull request by bxf12315**
… fine

@cristianonicolai @nmirasch Please review it.

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
@bxf12315 issue still present for me, either for loading the process instances at first time ( State Active not checked ) and also when loading one of the saved filters, State is not checked.

**Comment by jstastny-cz** [comment]
@bxf12315 I am seeing the same as @cristianonicolai .

**Comment by bxf12315** [comment]
@cristianonicolai @jstastny-cz I can produce it on our downstream zip. But I rebuild my local business-central-parent and test it, it is working fine. So I rebase it on the master branch, please re-full-downstream build it.

**Comment by jstastny-cz** [comment]
Jenkins execute full downstream build

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 1]
```diff
@@ -192,9 +191,16 @@ protected void addSLAComplianceFilter() {
...
+        if (activeFilterItem.getKey().equals(constants.State()) &&
+                activeFilterItem.getValue() instanceof List) {
             final List<String> values = (List<String>) activeFilterItem.getValue();
             values.forEach(v -> view.checkSelectFilter(constants.State(), String.valueOf(v)));
         }
+
+        if (activeFilterItem.getKey().equals(COLUMN_ERROR_COUNT) &&
+                activeFilterItem.getValue() instanceof List) {
+            final List<String> values = (List<String>) activeFilterItem.getValue();
+            values.forEach(v -> view.checkSelectFilter(constants.Errors(), (String.valueOf(v).equals("0") || String.valueOf(v) == "0") ? "true" : "false"));
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: again equals for string is enough, no need to compare identity aferwards.

Comment (jstastny-cz) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: plus v variable is already a String.

Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]:
```suggestion
            values.forEach(v -> view.checkSelectFilter(constants.Errors(), Boolean.valueOf("0".equals(v)).toString());
```

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenterTest.java [hunk 2]
```diff
@@ -107,4 +105,19 @@ public void onActiveFilterAddedTest() {
...
         verify(getView()).checkSelectFilter(Constants.INSTANCE.State(), String.valueOf(ProcessInstance.STATE_COMPLETED));
     }
+
+    @Test
+    public void testOnActiveFilterAdded() {
+        ActiveFilterItem testActiveFilterItem = new ActiveFilterItem(Constants.INSTANCE.State(), "test lable", null, Arrays.asList("Active"), null);
+        presenter.onActiveFilterAdded(testActiveFilterItem);
+        verify(getView()).checkSelectFilter(Constants.INSTANCE.State(), "Active");
+
+        ActiveFilterItem activeWithErrorsFilterItem = new ActiveFilterItem("errorCount", "test lable", null, Arrays.asList("0"), null);
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenterTest.java]: lable->label

**Comment by jstastny-cz** [comment]
Please see the comments inline.

The issue itself seems to be resolved.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java (hunk 1 above)
Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: when we use -Dfull build compare two strings(that's number like "1"), equals is not working.so I add "==" logic it. For my local test(gwt run), must be use equals. I know it's a little weird.

Comment (cristianonicolai) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: +1

**Comment by bxf12315** [comment]
@jstastny-cz @cristianonicolai updated

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/basic/BasicFiltersViewImpl.java [hunk 3]
```diff
@@ -313,7 +314,7 @@ public void checkSelectFilter(final String label,
         selectInputs.computeIfPresent(label,
                                       (key, values) -> {
                                           values.forEach(i -> {
-                                              if (i.getValue().equals(value) && i.getChecked() == false) {
+                                              if ((i.getValue().equals(value)) && i.getChecked() == false) {
```
Comment (cristianonicolai) [inline jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/basic/BasicFiltersViewImpl.java]: is this needed?

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 4]
```diff
@@ -192,9 +191,16 @@ protected void addSLAComplianceFilter() {
...
+                activeFilterItem.getValue() instanceof List) {
             final List<String> values = (List<String>) activeFilterItem.getValue();
-            values.forEach(v -> view.checkSelectFilter(constants.State(), String.valueOf(v)));
+            values.forEach(v -> view.checkSelectFilter(constants.State(), v));
+        }
+
+        if (activeFilterItem.getKey().equals(COLUMN_ERROR_COUNT) &&
+                activeFilterItem.getValue() instanceof List) {
+            final List<String> values = (List<String>) activeFilterItem.getValue();
+            values.forEach(v -> view.checkSelectFilter(constants.Errors(), Boolean.valueOf(v.equals("0")).toString()));
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: if you need to change the value, why not do it when you create the ActiveFilterItem as you already map coreFunctionFilter.getParameters()?

Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]:
```suggestion
            values.forEach(v -> view.checkSelectFilter(constants.Errors(), Boolean.valueOf(v).toString()));
```

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: error's ActiveFilterItem's parameters are "0" or "1",but selectInput's name's "true" and "false.". but state's ActiveFilterItem's parameters are "0" or "1"., it's selectInput's name also is "0","1"....
The logic in here :https://github.com/kiegroup/jbpm-wb/pull/1377/files#diff-d94e7092bb936da858898a7c5d50c2baR92

**Comment by bxf12315** [comment]
@cristianonicolai updated.

**Comment by jstastny-cz** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
@bxf12315 there are test failures now, please review.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Pull request merged**